python/tk_multi_publish2/api/task.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import uuid
+
+import sgtk
+from .plugins import Setting, PublishPluginInstance
+
+logger = sgtk.platform.get_logger(__name__)
+
+
+class PublishTask(object):
+    """
+    A task is a particular unit of work which can to be carried by the
+    publishing process. A task can be thought of as a 'plugin instance', or a
+    particular publishing plugin operating on a collected item.
+    """
```

Comment:
An example of the serialized version would be nice.

Reply:
same comment about making the serialized state public.